shell/cc-window.c

```diff
@@ -1369,6 +1370,17 @@ update_small_screen_settings (CcWindow *self)
 
   self->priv->small_screen = small;
 
+  /* If we're on a big screen, the icon view shall
+   * limit the number of columns to 6. Otherwise,
+   * make it adapt to small screen sizes.
+   */
+  item_views = get_item_views (self);
+
+  for (l = item_views; l != NULL; l = l->next)
+    gtk_icon_view_set_columns (l->data, small == SMALL_SCREEN_TRUE ? -1 : 6);
```

Comment:
This could probably be squashed with the other commit that removes the 6 columns.